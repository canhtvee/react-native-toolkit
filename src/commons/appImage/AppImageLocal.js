import React from 'react';
import {Image} from 'react-native';
import {getResourceImage} from '../../utils';
import {styles} from './styles';

export function AppImageLocal({name, ...props}) {
  return (
    <Image source={getResourceImage(name)} style={styles.image} {...props} />
  );
}
