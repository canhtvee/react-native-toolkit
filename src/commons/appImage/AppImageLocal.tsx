import React from 'react';
import {Image, ImageProps} from 'react-native';
import {getResourceImage, ResourceImageNameType} from '../../utils';
import {styles} from './styles';

export interface AppImageLocalProps extends Omit<ImageProps, 'source'> {
  name: ResourceImageNameType;
}

export function AppImageLocal({name, ...props}: AppImageLocalProps) {
  return (
    <Image source={getResourceImage(name)} style={styles.image} {...props} />
  );
}
