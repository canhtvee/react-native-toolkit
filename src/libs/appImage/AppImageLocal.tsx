import React from 'react';
import {Image, ImageProps} from 'react-native';
import {getResourceImage, ResourceImageNameType} from '../../utils';

export interface AppImageLocalProps extends Omit<ImageProps, 'source'> {
  name: ResourceImageNameType;
}

export function AppImageLocal({name, ...props}: AppImageLocalProps) {
  return <Image source={getResourceImage(name)} {...props} />;
}
