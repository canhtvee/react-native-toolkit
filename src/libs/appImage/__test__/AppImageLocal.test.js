import React from 'react';
import {View} from 'react-native';
import {render, cleanup} from '@testing-library/react-native';

import {AppImageLocal} from '../AppImageLocal';

/**
 * Run test command
 * yarn test -u -t="AppImageLocal"
 */

afterAll(cleanup);

describe('AppImageLocal', () => {
  test('should render correctly', () => {
    const tree = render(
      <View>
        {/* <AppImageLocal
          name="app_logo"
          style={{
            backgroundColor: 'red',
            margin: 20,
          }}
        /> */}
      </View>,
    );
    expect(tree).toMatchSnapshot();
  });
});
