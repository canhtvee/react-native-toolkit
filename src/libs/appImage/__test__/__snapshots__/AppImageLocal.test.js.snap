// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppImageLocal should render correctly 1`] = `<View />`;
